src/aliases.ts

```diff
@@ -0,0 +1,26 @@
+import * as Config from "./config"
+
+export function commandIsAlias(command: string): boolean {
+    return Config.get("exaliases", command) !== undefined
+}
+
+export function expandExstr(exstr: string): string {
+    // Split on whitespace
+    const [command, ...args] = exstr.trim().split(/\s+/)
+    const expandedCommand = getAliasExpandRecur(command)
+
+    // Replace only replaces first occurence by default
+    const expandedExstr = exstr.replace(command, expandedCommand)
+    return expandedExstr
+}
+
+export function getAliasExpandRecur(command: string): string {
+    // Base case: alias not found; return original command
+    if(!commandIsAlias(command)) {
+        return command
+    }
+
+    // Alias exists; expand it recursively
+    const expanded = Config.get("exaliases", command)
+    return getAliasExpandRecur(expanded)
```

Comment:
Bad aliases can cause an infinite loop here.

e.g. `alias t t` or `alias a b; alias b a`.

Can avoid this by storing a list of each alias expanded so far and throwing an exception if that list ever has a duplicate in it.